src/main.c

```diff
@@ -496,8 +497,15 @@ int pr_cmd_read(cmd_rec **res) {
     if (cmd_buflen < 0) {
       if (errno == E2BIG) {
         /* The client sent a too-long command which was ignored; give
-         * them another chance?
+         * them a few more chances, with minor delays?
          */
+        too_large_count++;
+        pr_timer_usleep(250 * 1000);
```

Comment:
This "sleep" helps mitigate the tightly-spinning loop, which is the core of the denial-of-service behavior.  The `too_large_count` limit limits the number of loops before the client is disconnected.